ssp

```diff
@@ -227,12 +222,14 @@ sub run {
 
     init();
 
-    Getopt::Long::GetOptionsFromArray(
-        \@argv,
-        'bugreport' => \$only_bugreport,
-        'csi'       => \$only_csi,
-        'timeout=i' => \$OPT_TIMEOUT,
-    );
+    if ( defined &Getopt::Long::GetOptionsFromArray ) {    # GetOpt::Long 2.36 and later only, Perl 5.8.8 on CentOS 5.11 includes 2.35
```

Comment:
So you effectively can't have argument parsing on perl 5.8.8?

Would it not be easier to just do Getopt::Long::GetOptions instead, as the array we're passing in here is already argv? IIRC, GetOptions existed even on GetOpt::Long 2.35?

Reply:
This is just a reversion of recent changes, it was this way for a long time.  Fixing it to work correctly on CentOS 5 may not be worth the effort, I don't know.  The only downside at this time is you aren't able to pass any options to SSP on CentOS 5 which would be a very rare occurrence.  Compared to it not working at all, it's still an improvement.

Reply:
Probably something more primitive would be better, at least then you can drop the dependency on Getopt::Long entirely:
`for ( my $i = 0; $i < scalar(@argv); $i++ ) { 
    # Check if we start with a '-'
    next if index( $argv[$i], '-' ) != 0;
    my ( $undashed ) = $argv[$i] =~ m/-+([a-z]+)/;
    $bugreport = 1 if $undashed eq 'bugreport';
    $csi = 1 if $undashed eq 'csi';
    $timeout = $argv[$i + 1] if $undashed eq 'timeout';
}`

Reply:
Thanks, something like this could work.  SSP doesn't (currently) have intense option handling requirements so GetOpt::Long is probably not a requirement. I've created TECH-477 for this.